### PR TITLE
Indexed properties in the Page Builder API

### LINKED ISSUE #616: Support indexed properties in Page Builder
Some properties are indexed. The syntax is *property_name[index]*. 
This of course cannot be a parameter name, from the Page Builder API.

We need to invent a way to express this anyway. `property_(index, ...)`?

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolves #616

The updated regular expression will handle double underscore followed by numbers and characters: `example__abc123`

These will have the same result
```
page = """
<|{data}|chart|mode=lines|x=Date|y[1]=Temp°C|y[2]=Min|y[3]=Max|line[1]=dash|color[2]=blue|color[3]=red|>
"""
```
```
with tgb.Page() as tgb_page:
    tgb.chart("{data}", mode="lines", x="Date", y__1="Temp°C", y__2="Min", y__3="Max", line__1="dash",color__2="blue", color__3="red")
```

### CODE REVIEW

File: taipy/gui/builder/_element.py [hunk 1]
```diff
@@ -28,6 +29,7 @@ class _Element(ABC):
 
     _ELEMENT_NAME = ""
     _DEFAULT_PROPERTY = ""
+    __RE_INDEXED_PROPERTY = re.compile(r"^(.*?)_([\d]+)$")
```
Comment (FredLL-Avaiga) [inline taipy/gui/builder/_element.py]: No need for the square brackets

**Review by FredLL-Avaiga — DISMISSED** [review]
Joyeux Noël ☃️

File: taipy/gui/builder/_element.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/builder/_element.py]: The square brackets is used for the declaration of the allowed chars, in this case numbers and is not included in the allowed chars

File: taipy/gui/builder/_element.py [hunk 2]
```diff
@@ -28,6 +29,7 @@ class _Element(ABC):
 
     _ELEMENT_NAME = ""
     _DEFAULT_PROPERTY = ""
+    __RE_INDEXED_PROPERTY = re.compile(r"^(.*?)(__([\w\d]+)|_([\d]+))$")
```
Comment (FredLL-Avaiga) [inline taipy/gui/builder/_element.py]: Should we need the 2 ?
ie have only 2 underscores as separator...

**Review by FredLL-Avaiga — DISMISSED** [review]

**Review by FabienLelaquais — APPROVED** [review]
As easy as your ABC.
Well done!

**Pull request merged**